### PR TITLE
Added docstring for ->

### PR DESCRIPTION
**Pull request by aramirezreyes**
Added docstring for `->`. Regarding #40424.

### CODE REVIEW

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -602,6 +602,25 @@ the last expression in the function body.
 """
 kw"function"
 
+"""
+    ->
+
+`->` is used to define anonymous functions:
```
Comment (fredrikekre) [inline base/docs/basedocs.jl:608]: Maybe something like this?
```suggestion
    x -> y

Create an anonymous function mapping argument(s) `x` to the function body `y`.
```

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -602,6 +602,25 @@ the last expression in the function body.
...
+
+```julia
+x -> x^2 + 2x - 1
+```
```
Comment (fredrikekre) [inline base/docs/basedocs.jl:612]:
```suggestion
```jldoctest
julia> f = x -> x^2 + 2x - 1
#1 (generic function with 1 method)

julia> f(2)
7
```
```

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -602,6 +602,25 @@ the last expression in the function body.
...
+A similar function can be defined with the short form notation:
+
+```julia
+function (x)
+    x^2 + 2x - 1
+end
+```
```
Comment (fredrikekre) [inline base/docs/basedocs.jl:619]: This doesn't seem to be necessary in the docstring for `->`?
```suggestion
```

**Review by fredrikekre — COMMENTED** [review]
Maybe it would be nice to show another example with multiple arguments too?

File: base/docs/basedocs.jl (hunk 2 above)
Comment (mcabbott) [reply base/docs/basedocs.jl:612]: Maybe worth showing a function with multiple arguments?
```
julia> map((x,y) -> f(x) * y, [2, 2, 2], 0:2)
3-element Vector{Int64}:
  0
  7
 14
```

**Comment by aramirezreyes** [comment]
Thanks for the guidance! I can make the changes a little later today.

**Pull request merged**